### PR TITLE
Handle parse-only agent blocks in client migrator

### PR DESCRIPTION
**Pull request by ReneWerner87**
## Summary
- add support for migrating AcquireAgent blocks that only parse requests into client calls
- ensure client import is added for the new migration path and adjust rewrite logic accordingly
- cover the new scenario with a dedicated test case

## Testing
- make lint
- make test

------
[Codex Task](https://chatgpt.com/codex/tasks/task_e_693358e669708326b4176da528789714)

## Summary by CodeRabbit

* **New Features**
  * Migration rewrites now track and propagate request bodies and default error names, and inject conditional error-handling scaffolding when needed.

* **Bug Fixes**
  * Rewrites preserve headers, methods and additional lines, avoid error-name collisions, and only mutate error handling when legacy patterns are detected; responses emit a single error field on failure.

* **Tests**
  * Expanded coverage for body/header propagation, client-request/send migrations, legacy-error scenarios, collision avoidance, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: cmd/internal/migrations/v3/client_usage_test.go [hunk 1]
```diff
@@ -929,6 +1002,258 @@ func handler(code string) {
...
+	cmd := newCmd(&buf)
+	require.NoError(t, v3.MigrateClientUsage(cmd, dir, nil, nil))
+
+	content := gofmtSource(t, readFile(t, file))
+	expected := gofmtSource(t, `package main
+
+import (
+    "fmt"
+
+    "github.com/gofiber/fiber/v3/client"
```
[the remark this answers is not in the document]
Comment (ReneWerner87) [reply cmd/internal/migrations/v3/client_usage_test.go]: wrong, this import is needed
